### DISCUSSION TITLE
TypeError: this.log is not a function

### DISCUSSION THREAD

**Discussion by harshsinghatz** [mswjs/msw#1523] [Q&A]
This is the exact error:
![image](https://user-images.githubusercontent.com/51085015/213637416-5cc76706-4a62-4aed-8ec5-e557f984b088.png)

The failing test basically tests a react component with redux . I am not able to reproduce the error as this error occurs randomly. Running the individual file there are no issues but when it runs with all the other test suites this test file fails.

Does this error sounds familiar? It would be helpful if you can speculate why this error might occur? 

Node version: 14.9.3

**Comment by harshsinghatz** [comment]
@kettanaito @mattcosta7 ?

**Comment by kettanaito** [comment]
Hey, @harshsinghatz. Thanks for raising this. 

No, the error doesn't look familiar. It seems to be related to the bound `debug` instance in that interceptor. Given it works everywhere else, could you share a reproduction repository with this? We've got a reproduction-only policy when looking into issues to save maintainers' time and help you resolve any unexpected behaviors much more efficiently. Thanks.

**Reply by mattcosta7** [reply]
I wonder if it's interesting that this happens in `XMLHttpRequestOverride.abort` - could be this `log` call happens after something else has been cleaned up?  but definitely strange its intermittent!

**Comment by tortilaman** [comment]
I have seen the same error with what I assume is a different setup:

<img width="847" alt="image" src="https://github.com/mswjs/msw/assets/5018268/04b4b473-23b2-4469-8510-9a4244f9306a">

---

I understand I haven't provided a reproduction, so this likely won't get maintainer bandwidth, but wanted to surface this in the interest of at least showing the above poster isn't an isolated instance. Again, this is an intermittent failure, that only occurs when the entire test suite is run. It's possible I'll have time to create a reproduction or investigate further later, if so I will provide an update.

**Reply by kettanaito** [reply]
Thanks for contributing to the discussion, @tortilaman. Please, if you get to making a reproduction repo, use `msw@next` instead. We've fixed a ton of issues in that `next` release tag and the nature of the current issues doesn't sound critical to be addressed in 1.0. Thanks.
